### PR TITLE
Make FlowOperation Callable

### LINKED ISSUE #308: Make FlowOperation callable
Currently the FlowOperations work differently than FlowCmdOperations. While the latter store their command internally and can be directly evaluated, the FlowOperations require lookup through the `_operation_functions` attribute of the FlowProject. We should change this and make FlowOperation callable and have it evaluate the operation function. Removing this indirection will centralize the logic for the execution directly in BaseFlowOperation and its subclasses, making it easier to modify for future changes since we won't have to check internally whether a given operation is a FlowOperation or a FlowCmdOperation.

### PR DESCRIPTION
**Pull request by kidrahahjo**
## Description

Removes the internal logic:
```
if isinstance(op, FlowCmdOperation):
    func = op._cmd
else:
    func = self._operation_functions[op_name]
```

Replace the above with:
```
if isinstance(op, FlowCmdOperation):
    func = op._cmd
else:
    func = op._op_func
```
## Motivation and Context

Resolves #308 

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

**Comment by kidrahahjo** [comment]
@vyasr this doesn't remove the instance check. Since we have two different variable names `_cmd` and `_op_func`, we can either use a try-except block or do the instance check. I think the instance check is more meaningful here. Please suggest.

File: flow/project.py [hunk 1]
```diff
@@ -3487,7 +3488,7 @@ def _main_exec(self, args):
             jobs = self
         try:
             try:
-                operation_function = self._operation_functions[args.operation]
+                operation_function = self._operations[args.operation]
             except KeyError:
                 operation = self._operations[args.operation]
```
Comment (b-butler) [inline flow/project.py:3493]: There is no need for the `try` `except` block any more.

Comment (kidrahahjo) [reply flow/project.py]: @b-butler I removed the block.

**Comment by kidrahahjo** [comment]
@b-butler @atravitz @lyrivera 
I am not sure why this check is there, should I remove it?
https://github.com/glotzerlab/signac-flow/pull/326/files#diff-896014ecae7b1581f5b837d124cdb4c9R3502

File: flow/project.py [hunk 2]
```diff
@@ -2254,8 +2256,8 @@ def _execute_operation(self, operation, timeout=None, pretend=False):
             operation.directives.get('fork', False)
             # Separate process needed to cancel with timeout:
             or timeout is not None
-            # The operation function is not registered with the class:
-            or operation.name not in self._operation_functions
+            # The operation function is of an instace of FlowCmdOperation:
```
Comment (atravitz) [inline flow/project.py:2259]:
```suggestion
            # The operation function is of an instance of FlowCmdOperation:
```

**Comment by vyasr** [comment]
> @vyasr this doesn't remove the instance check. Since we have two different variable names `_cmd` and `_op_func`, we can either use a try-except block or do the instance check. I think the instance check is more meaningful here. Please suggest.

That's fine. This at least removes the indirection required to get operation functions from the project. Let's keep these types of PRs small, then you'll be able to get the merged faster :)

**Pull request merged**